HRIS.Services.Tests/Services/EmployeeDocumentServiceUnitTest.cs

```diff
@@ -309,10 +472,79 @@ public async Task UpdateEmployeeDocumentFail()
                         _employeeDocumentService.UpdateEmployeeDocument(EmployeeDocumentTestData.EmployeeDocumentApproved.ToDto(), "test@retrorabbit.co.za"));
     }
 
+    [Fact]
+    public async Task UpdateEmployeeDocument_UnauthorizedAccess()
+    {
+        _identity.Setup(i => i.Role).Returns("Employee");
+        _identity.SetupGet(i => i.EmployeeId).Returns(2);
+
+        _unitOfWorkMock.Setup(x => x.EmployeeDocument.Any(It.IsAny<Expression<Func<EmployeeDocument, bool>>>()))
+        .ReturnsAsync(true);
+
+        _unitOfWorkMock.Setup(x => x.EmployeeDocument.Update(It.IsAny<EmployeeDocument>()))
+        .ReturnsAsync(EmployeeDocumentTestData.EmployeeDocumentPending);
+
+        var exception = await Assert.ThrowsAsync<CustomException>(() =>
+           _employeeDocumentServiceAuthNo.UpdateEmployeeDocument(EmployeeDocumentTestData.EmployeeDocumentPending.ToDto(), "test@retrorabbit.co.za"));
+
+        Assert.Equivalent("Unauthorized Access.", exception.Message);
+
+        _unitOfWorkMock.Verify(x => x.EmployeeDocument.Any(It.IsAny<Expression<Func<EmployeeDocument, bool>>>()),
+            Times.Once);
+        _unitOfWorkMock.Verify(x => x.EmployeeDocument.Update(It.IsAny<EmployeeDocument>()), Times.Never);
+    }
+
+    [Fact]
+    public async Task UpdateEmployeeDocument_EmployeeNotFound()
+    {
+        var employeeDocumentDto = EmployeeDocumentTestData.EmployeeDocumentPending.ToDto();
+
```

Comment:
Refactor: see if you can make _employeeDocumentDto initialized in the constructor, have a look at the other variables repeated. All of them can be init. in the constructor so we do not repeat code and the lines of code is reduced.

Reply:
I will have a look at what duplicates I can remove and init in the constructor

Reply:
Removed duplicates and initialised in the constructor done